test/tstKokkosToolsAnnotations.cpp

```diff
@@ -0,0 +1,181 @@
+/****************************************************************************
+ * Copyright (c) 2012-2020 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include "ArborX_EnableDeviceTypes.hpp" // ARBORX_DEVICE_TYPES
+#include <ArborX_LinearBVH.hpp>
+
+#include <impl/Kokkos_Profiling.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+#include <string>
+
+#include "Search_UnitTestHelpers.hpp"
+
+BOOST_AUTO_TEST_SUITE(KokkosToolsAnnotations)
+
+namespace tt = boost::test_tools;
+
+template <typename T>
+struct TreeTypeTraits;
+
+template <typename... DeviceTypes>
+struct TreeTypeTraits<std::tuple<DeviceTypes...>>
+{
+  using type = std::tuple<ArborX::BVH<DeviceTypes>...>;
+};
+
+using TreeTypes = typename TreeTypeTraits<ARBORX_DEVICE_TYPES>::type;
+
+bool isPrefixedWith(std::string const &s, std::string const &prefix)
+{
+  return s.find(prefix) == 0;
+}
+
+BOOST_AUTO_TEST_CASE(is_prefixed_with)
+{
+  BOOST_TEST(isPrefixedWith("ArborX::Whatever", "ArborX"));
+  BOOST_TEST(!isPrefixedWith("Nope", "ArborX"));
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(bvh_bvh_allocations_prefixed, Tree, TreeTypes)
+{
+  Kokkos::Tools::Experimental::set_allocate_data_callback(
+      [](Kokkos::Profiling::SpaceHandle /*handle*/, const char *label,
+         void const * /*ptr*/, uint64_t /*size*/) {
+        std::cout << label << '\n';
+        BOOST_TEST(
+            (isPrefixedWith(label, "ArborX::BVH::BVH::") ||
+             isPrefixedWith(label, "ArborX::Sorting::") ||
+             isPrefixedWith(label,
+                            "Kokkos::Serial::") || // unsure what's going on
```

Comment:
This was coming from https://github.com/kokkos/kokkos/blob/75e7b7aa627b60ea289cf7b88cc49b53ed36bd35/core/src/impl/Kokkos_Serial.cpp#L121

Reply:
I could try to make an initial allocation that's big enough